### PR TITLE
113 create make table 32 gt

### LINKED ISSUE #113: Create `make_table_32_gtsum`
Create function `make_table_32_gtsum` to build table 32 using the `gt` and `gtsummary` packages.
See #17 and https://pharmaverse.github.io/falcon/quarto/table-templates/template-table_32.html for specifications for this table.

The table output should be very similar to the output created by `make_table_32`.

The Table 32 template page on the falcon site will be updated once the function is ready.

### PR DESCRIPTION
**Pull request by aassuied-ps**
Closes #113

### CODE REVIEW

**Comment by ayogasekaram** [comment]
Hey @aassuied-ps great work! I made a few corrections and pushed those changes. I think the PR is missing an update to the quarto page for table 32 and a snapshot in the test file. We can work on those next week.

File: quarto/table-templates/template-table_32.qmd [hunk 1]
```diff
@@ -40,6 +40,64 @@ Required variables:
 -   **`advs`**: `USUBJID`, `AVISITN`, `PARAMCD`, `AVAL`, `AVALU`, and the variables specified by `arm_var` and `saffl_var`.
 -   **`alt_counts_df`** (if specified): `USUBJID` and the variables specified by `arm_var` and `saffl_var`.
 
+<<<<<<< HEAD
+<<<<<<< HEAD
+| Argument         | Description                                                                                                                                                                                                                                            | Default      |
+|:--------------|:------------------------------------------|:--------------|
+| `advs`           | (`data.frame`) Dataset (typically ADVS) required to build table.                                                                                                                                                                                       | *No default* |
```
Comment (ayogasekaram) [inline quarto/table-templates/template-table_32.qmd:47]: The function requires the `adsl` dataset as well I believe. So this should be included.

File: quarto/table-templates/template-table_32.qmd [hunk 2]
```diff
@@ -40,6 +40,64 @@ Required variables:
...
+------------------------------------------------------------------------
+
+Required variables:
+
+-   **`advs`**: `SAFFL`, `USUBJID`, `AVISITN`, `PARAMCD`, `AVAL`, `AVALU`, and the variables specified by `arm_var`.
+-   **`alt_counts_df`** (if specified): `SAFFL`, `USUBJID`, and the variable specified by `arm_var`.
+
+| Argument        | Description                                                                                                        | Default              |
+|:-----------------|:----------------------------------|:-----------------|
+| `advs`          | (`data.frame`) Dataset (typically ADVS) required to build table.                                                   | *No default*         |
```
Comment (ayogasekaram) [inline quarto/table-templates/template-table_32.qmd:94]: adsl included here too

Comment (aassuied-ps) [reply quarto/table-templates/template-table_32.qmd:94]: The current version of template-table_32.qmd includes alt_counts_df on lines 48 and 95, the same way as table 6 or others. Should it be different here?

I'm gonna modify the branch also to include saffl_var to the gt table.

Comment (ayogasekaram) [reply quarto/table-templates/template-table_32.qmd:94]: But the `gtsum` function [currently requires](https://github.com/pharmaverse/falcon/blob/33d76e8b737c8de601fd8c016e20a6017121cbc7/R/fda-table_32.R#L124-L130) the adsl data (apart from the alt_counts_df). Should the function be reworked then to remove the adsl processing?

Comment (aassuied-ps) [reply quarto/table-templates/template-table_32.qmd:94]: I misunderstood that point, sorry. I'm gonna find a way to add adsl parameter to the documentation, or think of a way to remove the adsl processing (if possible - the issue is on the population count for percentages).

**Comment by ayogasekaram** [comment]
Hey @aassuied-ps. I pushed a few changes which should allow the PR to pass the GH checks. I also ended up removing the section asking for the `adsl` pop. I reran the tests and I don't think that df is necessary - with these changes, it's very similar to the main `make_table_32` function.

**Review by edelarua — APPROVED** [review]
I updated the table header label and removed the `alt_counts_df` parameter that currently isn't doing anything. Should be ok to merge now! (might want to update tests again)

**Comment by edelarua** [comment]
Can you also update the table template page to remove the `alt_counts_df` parameter before merging? That should fix the failing checks

**Comment by aassuied-ps** [comment]
All checks passed this time.

I also modified the function to delete AVALU from the final dataset because AVALU was described as well.

table_32_gtsummary_1

**Pull request merged**